gateway/test/crud_test.go

```diff
@@ -3554,6 +3554,23 @@ func (s *GatewayOpsTestSuite) TestMutateIn() {
 		})
 	})
 
+	s.Run("PathExistsWithInsertSemantics", func() {
```

Comment:
This test is validating the PathExists case rather than the DocExists case.  Although, it's possible we may have been missing a PathExists test as well?

Reply:
https://github.com/couchbase/protostellar/blob/master/couchbase/kv/v1/kv.proto#L388
This is the 'Insert Semantics' we are talking about.

Reply:
I did confirm that we already have a test for Insert specs:
https://github.com/couchbase/stellar-gateway/blob/39f357cf0225b43f4197a26d1e13d7cd1a57d32c/gateway/test/crud_test.go#L3245-L3248